### ISSUE TITLE
upstream/master branch and "forked" master branch behaves same

### ISSUE THREAD

**Issue by ugur1yildiz** [desktop/desktop#10898]
### Describe the bug

While creating a branch on a forked repository although indicating based on default master branch it is created on upstream/master.

### Version & OS

Version 2.5.6 on Windows 10 x64

### Steps to reproduce the behavior

1. Go to Branch menu item
2. Selecting New Branch ...
3. Giving a name to new branch and pressing OK

### Expected behavior

New branch should be created basing the forked master branch.

### Actual behavior

It is created basing upstream/master

### Screenshots

Add screenshots to help explain your problem, if applicable.

### Logs

2020-10-21T22:19:53.263Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: edx-dl - ugur1yildiz (has token)
2020-10-21T22:19:55.030Z - info: [ui] [Timing] Action 'checkout branch from list' for 'rjfeddeler/edx-dl' took 1.766s
2020-10-21T22:20:11.893Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: edx-dl - ugur1yildiz (has token)
2020-10-21T22:20:13.361Z - info: [ui] [Timing] Action 'create branch' for 'rjfeddeler/edx-dl' took 1.810s

2020-10-21T22:37:22.516Z - error: [ui] `git -c credential.helper= -c protocol.version=2 checkout --progress upstream/master -b master --` exited with an unexpected code: 128.
stderr:
fatal: A branch named 'master' already exists.

(The error was parsed as 26: A branch with that name already exists.)

### Additional context

Even if tried to rebase the newly created branch on forked master it indicates that this branch is up to date with forked master.
But actually not, upstream/master and forked master are different.

Moreover when try to change current branch from the drop-down list it gives error as "A branch with that name already exists." See in last part of the log.

**Comment by steveward** [comment]
Thanks for the report @ugur1yildiz. Can you run the following command from the command line and let me know the full output?

`git branch -a -vv`

**Comment by ugur1yildiz** [comment]
In Gui when I try to switch from master branch to remotes/upstream/master (denoted as upstream/master in gui) it gives error as "A branch with that name already exists."

Here is the list.

```
* master                                   56b90a8 [origin/master] display progress with tqdm
  remotes/origin/HEAD                      -> origin/master
  remotes/origin/fix-0-courses-available   b6c77a6 test_parsing: Adapt to new structure of course pages.
  remotes/origin/master                    56b90a8 display progress with tqdm
  remotes/origin/new-python-in-travis      1b95893 travis: Test with Python 3.5 and 3.6
  remotes/origin/use-external-downloader   6d7de0f edx_dl/edx_dl: Dirty, WIP, not-to-be merged use of external downloader.
  remotes/upstream/fix-0-courses-available b6c77a6 test_parsing: Adapt to new structure of course pages.
  remotes/upstream/master                  e9a5e9c remove double quotes in twine credentials in the workflow
  remotes/upstream/new-python-in-travis    1b95893 travis: Test with Python 3.5 and 3.6
  remotes/upstream/use-external-downloader 6d7de0f edx_dl/edx_dl: Dirty, WIP, not-to-be merged use of external downloader.
```

**Comment by steveward** [comment]
@ugur1yildiz apologies for the delay. When you clone a fork in GitHub Desktop it gives you the option of choosing how you want to use the fork:

![image](https://user-images.githubusercontent.com/721500/99729458-f2103780-2a88-11eb-8fd8-db4db65e7232.png)


This option affects what default branch new branches are based off of. Do you remember what option you selected when cloning this repository? You can change this by going to `Repository` > `Repository Settings` > `Fork Behavior`.
